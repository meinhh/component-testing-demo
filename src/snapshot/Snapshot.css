.normal {
	background-color: blueviolet;
}

.hovered {
	background-color: blanchedalmond;
}