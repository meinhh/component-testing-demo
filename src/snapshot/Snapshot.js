import React, { Component } from 'react';
import InnerComponent from '../InnerComponent';

export default class Snapshot extends Component {
	constructor(props) {
		super(props);

		this.handleClick = () => { };
	}
	render() {
		return (
			<div>
				<h1 className="title">{this.props.title}</h1>
				<InnerComponent />
				<button onClick={this.handleClick}>CLICK ME</button>
			</div>
		);
	}
}