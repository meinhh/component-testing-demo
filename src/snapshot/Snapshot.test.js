import React from 'react';
import Snapshot from './Snapshot';
import renderer from 'react-test-renderer';

describe('snapshot tests', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Snapshot page="http://www.facebook.com">Facebook</Snapshot>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as an anchor when no page is set', () => {
		const tree = renderer.create(<Snapshot>Facebook</Snapshot>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('properly escapes quotes', () => {
		const tree = renderer
			.create(<Snapshot>{"\"Facebook\" \\'is \\ 'awesome'"}</Snapshot>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('changes the class when hovered', () => {
		const component = renderer.create(
			<Snapshot page="http://www.facebook.com">Facebook</Snapshot>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		tree.props.onMouseEnter();
		tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		tree.props.onMouseLeave();
		tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});