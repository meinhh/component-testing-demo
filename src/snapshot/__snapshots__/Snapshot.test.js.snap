// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`snapshot tests changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`snapshot tests changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`snapshot tests properly escapes quotes 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  "Facebook" \\'is \\ 'awesome'
</a>
`;

exports[`snapshot tests renders as an anchor when no page is set 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`snapshot tests renders correctly 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
